### PR TITLE
[XCUITests] Add iOS XCUITests to CI

### PR DESCRIPTION
**Pull request by joannaquu**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

Binary change:

| File | Before | After | Delta |
|------|--------|-------|-------|
| libFluentUI.a | 55,562,128 bytes | 55,562,128 bytes | 0 bytes |

1. Updated CI to run iOS XCUITests every PR. 
2. Changed tests to run in series. Initially, tests were executed in parallel. However, this caused nondeterministic failures. 
3. Simplified navigation logic in `setUpWithError`.

### Verification

Screenshot 2023-01-09 at 3 59 46 PM

### Pull request checklist

This PR has considered:
- [ ] Light and Dark appearances
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/1480)

### CODE REVIEW

**Comment by harrieshin** [comment]
your command line, are we building (again) and then testing per target? once we have the build ready, can we just run test without building? or are you just building the test code of fluent demo app?

**Comment by joannaquu** [comment]
> your command line, are we building (again) and then testing per target? once we have the build ready, can we just run test without building? or are you just building the test code of fluent demo app?

Going through the logs, I only see one build succeeded so I believe it's building once and then testing each target. I noticed that it's testing the tests in FluentUITests target and then the ui tests in the FluentUIDemoTests target.

**Review by harrieshin — APPROVED** [review]

**Review by lyzhan7 — APPROVED** [review]

File: ios/FluentUI.Demo/FluentUI.Demo.xcodeproj/xcshareddata/xcschemes/Demo.Development.xcscheme [hunk 1]
```diff
@@ -75,8 +75,7 @@
             </BuildableReference>
          </TestableReference>
          <TestableReference
-            skipped = "NO"
-            parallelizable = "YES">
```
Comment (lyzhan7) [inline ios/FluentUI.Demo/FluentUI.Demo.xcodeproj/xcshareddata/xcschemes/Demo.Development.xcscheme:79]: Just to check, removing parallelizable here was what fixed the tests? I wonder why this got added to iOS and not macOS

Comment (joannaquu) [reply ios/FluentUI.Demo/FluentUI.Demo.xcodeproj/xcshareddata/xcschemes/Demo.Development.xcscheme:79]: Yep! This was auto added when I first created the launch tests so I think there's a flag set somewhere but I'm looking into it.

Comment (joannaquu) [reply ios/FluentUI.Demo/FluentUI.Demo.xcodeproj/xcshareddata/xcschemes/Demo.Development.xcscheme:79]: Hm looking at the macOS scheme, it is also set to parallelizable so I'm wondering why that's not an issue there.

File: ios/FluentUI.Demo/FluentUI.Demo/DemoListViewController.swift [hunk 2]
```diff
@@ -71,7 +71,7 @@ class DemoListViewController: DemoTableViewController {
                 }
             }
 
-            DemoListViewController.isFirstLaunch = false
```
Comment (harrieshin) [inline ios/FluentUI.Demo/FluentUI.Demo/DemoListViewController.swift:74]: why even have this isFirstLaunch variable if we aren't going to use it?

**Pull request merged**